egregoria/src/map_interaction/itinerary.rs

```diff
@@ -10,9 +10,9 @@ use specs::Component;
 
 #[derive(Component, Debug, Default, Inspect, Serialize, Deserialize)]
 pub struct Itinerary {
-    kind: ItineraryKind,
+    pub kind: ItineraryKind,
     #[inspect(proxy_type = "InspectVec<Vec2>")]
-    local_path: Vec<Vec2>,
+    pub local_path: Vec<Vec2>,
```

Comment:
I prefer keeping code local, and did some effort to not make theses public. If it's not too much hassle I'd prefer having a constructor and keeping theses private as it makes invariant reasoning easier in general.

Reply:
After reading the entire code, you just need to make a "prepend_points" method, which I also say in another comment.

Reply:
I kinda disagree -- invariants are still preserved as long as we don't use `mut` on the Itinerary object. I also dislike excessive, OO-style wrapping, and I think it's just two ways to do one thing.